jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -466,21 +464,20 @@ protected void doStop() throws Exception
 
         MultiException mex = new MultiException();
 
-        try
+        if (getStopTimeout() > 0)
         {
-            // list if graceful futures
-            List<Future<Void>> futures = new ArrayList<>();
-            // First shutdown the network connectors to stop accepting new connections
-            for (Connector connector : _connectors)
+            try
             {
-                futures.add(connector.shutdown());
+                long end = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(getStopTimeout());
+                Graceful.shutdown(this).get(getStopTimeout(), TimeUnit.MILLISECONDS);
```

Comment:
I think this should be the only line in the try/catch, otherwise if it throws the stopTimeout will never be set on the ThreadPool.

Reply:
ack